service/storage_proxy.cc

```diff
@@ -5427,9 +5447,25 @@ storage_proxy::handle_read_digest(const rpc::client_info& cinfo, rpc::opt_time_p
                 }
                 auto timeout = t ? *t : db::no_timeout;
                 return p->query_result_local_digest(std::move(s), cmd, std::move(pr2.first), trace_state_ptr, timeout, da, rate_limit_info);
-            }).then_wrapped([this, &trace_state_ptr, src_ip] (future<rpc::tuple<query::result_digest, api::timestamp_type, cache_temperature>> f) mutable {
+            }).then_wrapped([this, &trace_state_ptr, src_ip] (future<rpc::tuple<query::result_digest, api::timestamp_type, cache_temperature, full_position>> f) mutable {
                 tracing::trace(trace_state_ptr, "read_digest handling is done, sending a response to /{}", src_ip);
-                return encode_replica_exception_for_rpc(std::move(f), [] { return std::make_tuple(query::result_digest(), api::missing_timestamp, cache_temperature::invalid()); });
+
+                using final_tuple_type = rpc::tuple<query::result_digest, long, cache_temperature, replica::exception_variant, full_position>;
+
+                if (!f.failed()) {
+                    auto&& [d, t, c, p] = f.get();
+                    return make_ready_future<final_tuple_type>(d, t, std::move(c), replica::exception_variant(), std::move(p));
+                }
+
+                std::exception_ptr eptr = f.get_exception();
+                if (features().typed_errors_in_read_rpc) {
```

Comment:
What happened to encode_replica_exception_for_rpc?

Reply:
It assumes the appended exception type is the very last in the result tuple, which is only true so long as we cannot change the order of elements in the tuple.

Reply:
@piodul please make a comment

Reply:
Maybe we could change encode_replica_exception_for_rpc so that it doesn't append the exception at the end of the tuple, but rather can insert it at a specified position?

Reply:
That would solve the problem.